File: 511_library/Film.cpp
#include "Film.h"

#include <iostream>
//#include <conio.h>



Film::Film (string call_number2,string title2, string subjects2, string year2, 
	string director2, string notes2):Media(call_number2, title2, subjects2, notes2)
{
	//call_number = call_number2;
	//title = title2;
	//subjects = subjects2;
	year= year2;
	director = director2;
	//notes = notes2;

}

void Film::display() const
{
	cout<<"FILM"<<endl;
	Media::display();
	cout<<"year   :"<<year<<endl;
	cout<<"Director   :"<<director<<endl;
}

bool Film::compare_other(const string& ss)
{
	//int i = -1;
	if (std::string::npos != notes.find(ss) || std::string::npos != director.find(ss) || std::string::npos != year.find(ss))
	{
		
			return true;
		
	}
	else
			return false;
	
	/*if (std::string::npos != director.find(ss))
	{
		if (i >= 0)
			return true;
		else
			return false;
	}

	if (std::string::npos != year.find(ss))
	{
		if (i >= 0)
			return true;
		else
			return false;
	}*/
}

File: Library_Python/Film.py
from Media import Media
class Film(Media):
    def __init__(self,callnumber1,title1,subjects1,year1,director1,notes1):
        super(Film,self).__init__(callnumber1,title1,subjects1,notes1)
        self.year=year1
        self.director=director1
    def display(self):
        print("Results")
        print("FILM")
        super(Film,self).display()
        print(self.year)
        print(self.director)
    def compare_other(self,string_other):
        if (self.notes.find(string_other) != -1 or self.director.find (string_other) != -1 or self.year.find(string_other) != -1):
            return True
        return False
